index.js

```diff
@@ -10,7 +10,7 @@ var has = Object.prototype.hasOwnProperty;
  * @api public
  */
 function querystring(query) {
-  var parser = /([^=?&]+)=([^&]*)/g
+  var parser = /([^=?&]+)=?([^&]*)?/g
```

Comment:
Yep, I love this, wasn't looking at empty string so didn't even try this.  Nice one.  Making the changes now.